src/twisted/internet/defer.py

```diff
@@ -927,6 +928,48 @@ def main(reactor):
     return coro
 
 
+def coroFnToDeferredFn(corofn):
```

Comment:
I'm not very fond of the name `coroFnToDeferredFn`. It is accurate, but awkward to remember and type. I don't have a better name though; I'll give it some thought.

I think it's worth adding type annotations: even though mypy's support for decorators is rather weak at the moment, it would at least force the body to be type-checked.

Reply:
Maybe since this will be always applied to a corotine function, there is no need to add "coroFo' to the name.

Maybe name it:

* `@defer.coroutineWraper`
* `@defer.coroutineWrap`
* `@defer.deferCoroutine`

Reply:
* `@defer.red`
* `@defer.redFn`

To mark the function as a red function

https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/

Reply:
Other options suggested on IRC:
- `@defer.deferred`
- `@defer.returnDeferred`
- `@defer.applyEnsureDeferred`

I think `returnDeferred` has a good balance between being short & simple and describing what the decorator does.

Reply:
```
@(lambda f: lambda *args, **kwargs: ensureDeferred(f(*args, **kwargs)))
```
for Python 3.9

Reply:
`@coroutineFunctionToDeferredFunction` is fine by me, but I'm not afraid of long names.

Looks like [in my code](https://github.com/burningmantech/ranger-ims-server/blob/master/src/ims/ext/trial.py#L36) I've called it `asyncAsDeferred`.

Reply:
`test_deferred.py` calls it `ensuringDeferred`.

I think it should be called `inlineCallbacks` though